test_generator/F2PConvertor.cpp

```diff
@@ -1,101 +1,174 @@
+    std::vector<std::string> two_dim_fields;
+};
+
+template <class T> struct tag { using type = T; };
+template <class Tag> using type = typename Tag::type;
+template <class T, size_t n>
+struct n_dim_vec : tag<std::vector<type<n_dim_vec<T, n - 1>>>> {};
+template <class T> struct n_dim_vec<T, 0> : tag<T> {};
+template <class T, size_t n> using n_dim_vec_t = type<n_dim_vec<T, n>>;
+
+template <size_t nbits, size_t es>
+void convert_float_to_posit(std::vector<double> &input,
+                            std::vector<unsigned long> &output) {
+    std::transform(input.cbegin(), input.cend(), output.begin(),
+                   [](float v) -> unsigned long {
+                       sw::unum::posit<nbits, es> p(v);
+                       return p.get().to_ulong();
+                   });
+}
+
+template <size_t nbits, size_t es>
+void convert_path(json_t json, std::string field, int dim,
+                  json_t &transformed) {
+    try {
+        switch (dim) {
+        case 1: {
+            auto data = json[field]["data"].get<n_dim_vec_t<double, 1>>();
+            std::vector<unsigned long> data_posit(data.size());
+            convert_float_to_posit<nbits, es>(data, data_posit);
+            transformed[field]["data"] = data_posit;
+        } break;
+        case 2: {
+            auto data = json[field]["data"].get<n_dim_vec_t<double, 2>>();
+            std::vector<std::vector<unsigned long>> data_posit(
+                data.size(), std::vector<unsigned long>(data[0].size(), 0));
+            for (int i = 0; i < data.size(); i++) {
+                convert_float_to_posit<nbits, es>(data[i], data_posit[i]);
+            }
+            transformed[field]["data"] = data_posit;
+        } break;
+        default:
+            std::cerr << "[Error] The value of the dimension should be 1 or 2"
+                      << std::endl;
+        }
+    } catch (nlohmann::json::type_error err) {
+        std::cerr << "[Error] Expected `" << field
+                  << ".data' field with type float[]" << std::endl;
+        exit(2);
+    }
 }
 
-json_t parse_data(int argc, char **argv) {
-    if (argc != 4)
-    {
-        print_helper();
+void convert_to_posit(json_t json, std::string field, int posit_width, int dim,
+                      json_t &transformed) {
+    switch (posit_width) {
+    case 8:
+        convert_path<8, 1>(json, field, dim, transformed);
+    case 16:
+        convert_path<16, 1>(json, field, dim, transformed);
+        break;
+    case 32:
+        convert_path<32, 2>(json, field, dim, transformed);
+        break;
+    case 64:
+        convert_path<64, 3>(json, field, dim, transformed);
+        break;
+    case 128:
+        convert_path<128, 4>(json, field, dim, transformed);
+        break;
+
+    default:
+        std::cerr << "[Error] The supported posit standards are 16, 32 and 64"
+                  << std::endl;
         exit(2);
     }
+}
+
+void split_strings(std::string input, char delimiter,
```

Comment:
A function `split_strings` should have the `vector<std::string>` as the return type. If you are going with reference, the non output arguments should be marked with `const`